backend/tracim_backend/views/core_api/workspace_controller.py

```diff
@@ -337,22 +338,23 @@ def create_workspaces_members_role(
                 email=hapic_data.body.user_email_or_public_name,
                 public_name=hapic_data.body.user_email_or_public_name
             )
-        except UserDoesNotExist:
-            try:
-                # TODO - G.M - 2018-07-05 - [UserCreation] Reenable email
-                # notification for creation
-                user = uapi.create_user(
-                    email=hapic_data.body.user_email_or_public_name,
-                    password=password_generator(),
-                    do_notify=True
-                )  # nopep8
-                newly_created = True
-                if app_config.EMAIL_NOTIFICATION_ACTIVATED and \
-                        app_config.EMAIL_NOTIFICATION_PROCESSING_MODE.lower() == 'sync':
-                    email_sent = True
-
-            except EmailValidationFailed:
-                raise UserCreationFailed('no valid mail given')
+        except UserDoesNotExist as exc:
+            if app_config.EMAIL_NOTIFICATION_ACTIVATED:
```

Comment:
The bes g way to write this code is :

```
if not notif:
    Raise exc

Try:
    Blablabla
Except:
    Blablabla
```

The difference are :

- if/else : put shortcase first (easier to read)
- one indentation level removed (less complexity)